### ISSUE TITLE
Accessibility: Picker - clicking on label does not open dropdown menu

### ISSUE THREAD

**Issue by malanirishi** [adobe/react-spectrum#2185]
# 🐛 Bug Report 

Click on label for a `Picker` implementation. The focus transfers to the field but dropdown menu does not open up. This is causing issues with accessibility.
It shows the correct behavior of opening dropdown menu with `ComboBox` if `menuTrigger="focus"` is added as prop.

## 🤔 Expected Behavior

Menu should open on clicking of label.

## 😯 Current Behavior

It does not open dropdown menu currently on clicking of label.

## 💁 Possible Solution

May need to add similar prop like `menuTrigger` as it is in `ComboBox`.

## 🔦 Context

Using it in Move Modals in MEUE application. `ComboBoxes` and `Pickers` both are used in the modals and their different behavior causes confusion and violates accessibility audit.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | latest
| Browser          | Chrome
| Operating System | MscOS Big Sur

## 🧢 Your Company/Team

Adobe/Marketo/MEUE

## 🕷 Tracking Issue

https://jira.corp.adobe.com/browse/HG-153243

**Comment by devongovett** [comment]
This behavior matches the behavior of a native `<select>` element - clicking on the label only focuses the select, it does not open the menu.

The ComboBox has the same behavior by default, unless `menuTrigger="focus"` is used as you found. Clicking the label is still just focusing the ComboBox, not directly opening the menu. But that makes sense in this case, because you've defined that the menu should open on focus.

I think this is working as expected.
